src/graphql/packet.rs

```diff
@@ -92,6 +100,44 @@ impl PacketQuery {
         )
         .await
     }
+
+    #[allow(clippy::unused_async)]
+    async fn pcap<'ctx>(&self, ctx: &Context<'ctx>, filter: PacketFilter) -> Result<Pcap> {
+        let db = ctx.data::<Database>()?;
+        let store = db.packet_store()?;
+        let mut packet_vector = Vec::new();
+        let mut pcap = Vec::new();
+        let mut key_prefix = Vec::with_capacity(filter.source.len() + TIMESTAMP_SIZE + 2);
+        key_prefix.extend_from_slice(filter.source.as_bytes());
+        key_prefix.push(0);
+        key_prefix.extend_from_slice(&filter.request_time.timestamp_nanos().to_be_bytes());
+        key_prefix.push(0);
```

Comment:
This part is mostly the same as in `packets`. Please, remove the repetition.

Reply:
For packet it's used with : 
```        query(
            after,
            before,
            first,
            last,
            |after, before, first, last| async move {
                load_connection(&store, &key_prefix, &filter, after, before, first, last)
            },
        )
        .await
```
And for Pcap:

```
        let iter = store.boundary_iter(
            &lower_closed_bound_key(&key_prefix, start),
            &upper_open_bound_key(&key_prefix, end),
            Direction::Forward,
        );
        let (records, _) = collect_records(iter, 1000, &filter)?;
```

I can make a function for key_prefix to remove repetition but I'm not sure about "db" and "store" since both those variables are used differently. How can I remove the repetition without changing both functions too much?

Reply:
Done

Reply:
Added a function called get_key_prefix_packets